### PR TITLE
Update Apache REEF and Checkstyle.

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
Apache REEF 0.15 is released at the maven central. We had better use 0.15.
This PR also updates Checkstyle with the same version of REEF,
and fixes some RedundantModifier errors accordingly.

This PR was tested by the following commands.
```bash
rm -rf ~/.m2/
mvn clean install
```

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Hi, @yunseong .
Could you review this when you have some time?

**Comment by dongjoon-hyun** [comment]
Hi, @jsjason .
Could you review this when you have some time?

**Comment by yunseong** [comment]
@dongjoon-hyun Thanks! On behalf of @jsjason, I'll test & merge it.

File: dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/data/NeuralNetworkDataParser.java [hunk 1]
```diff
@@ -139,7 +139,7 @@ public void parse() {
     private final List<Matrix> matrixList;
     private final List<Integer> labelList;
 
-    public BatchGenerator(final List<Pair<Pair<Matrix, int[]>, Boolean>> dataList,
+    BatchGenerator(final List<Pair<Pair<Matrix, int[]>, Boolean>> dataList,
                           final boolean isValidation) {
```
Comment (yunseong) [inline dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/data/NeuralNetworkDataParser.java]: nit: indent

File: dolphin-ps/src/main/java/edu/snu/dolphin/ps/worker/partitioned/PartitionedParameterWorker.java [hunk 2]
```diff
@@ -399,7 +399,7 @@ public V get() {
     private volatile boolean close = false;
     private volatile boolean shutdown = false;
 
-    public Partition(final ConcurrentMap<K, PullFuture<V>> pendingPulls,
+    Partition(final ConcurrentMap<K, PullFuture<V>> pendingPulls,
                      final ServerResolver serverResolver,
                      final InjectionFuture<PartitionedWorkerMsgSender<K, P>> sender,
                      final int queueSize,
```
Comment (yunseong) [inline dolphin-ps/src/main/java/edu/snu/dolphin/ps/worker/partitioned/PartitionedParameterWorker.java]: nit: indent

**Comment by yunseong** [comment]
@dongjoon-hyun Tests passed on my local machine as well. For the minor nitpicks, I was thinking to make changes by myself, but it turns out to be unachievable as the branch is at your forked repo. I'm totally fine with the other changes, so could you let me know once you're done? Thanks!

File: dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/data/NeuralNetworkDataParser.java (hunk 1 above)
Comment (dongjoon-hyun) [inline dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/data/NeuralNetworkDataParser.java]: Ooops. Thank you.

**Comment by dongjoon-hyun** [comment]
Thank you for review, @yunseong ! I'll update soon.

**Comment by dongjoon-hyun** [comment]
The PR is updated. Thank you, @yunseong .

**Comment by yunseong** [comment]
@dongjoon-hyun Looks good. I'll merge this. Thanks!

**Comment by dongjoon-hyun** [comment]
Thank you, @yunseong ! :)

**Pull request merged**